src/data-types/binary.ts

```diff
@@ -60,6 +60,17 @@ const Binary: { maximumLength: number } & DataType = {
     yield parameter.value.slice(0, parameter.length !== undefined ? Math.min(parameter.length, this.maximumLength) : this.maximumLength);
   },
 
+  toBuffer: function(parameter) {
+    const value = parameter.value as string | Buffer;
+
+    if (value != null) {
+      return Buffer.isBuffer(value) ? value : Buffer.from(value);
+    } else {
+      // PLP NULL
+      return Buffer.from([ 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF ]);
+    }
```

Comment:
I noticed that `toBuffer` always has special handling for `null` values, like here, where the PLP NULL buffer value is returned.

I also noticed that `toBuffer` is never called when the parameter value is null, because in those cases the `varbinary` column that actually holds the encrypted information will simply be set to null instead. 🤔

I think that needs to be verified, and if I'm correct, we can remove all the `null` handling logic from `toBuffer`.

Potentially, we can remove `toBuffer` as a whole and instead re-use `generateParameterData`.